### PR TITLE
SONARGO-40 Vulnerability imported through external report Golangci lint should be set to Security in Software Quality

### PR DESCRIPTION
**Pull request by mstachniuk**
[SONARGO-40](https://sonarsource.atlassian.net/browse/SONARGO-40)

Part of 

[SONARGO-40]: https://sonarsource.atlassian.net/browse/SONARGO-40?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by petertrr — APPROVED** [review]
LGTM!

**Comment by mstachniuk** [comment]
Added extra logic to not set impact for SonarQube Cloud. Inspiration: https://github.com/SonarSource/sonar-php/commit/2c2926fd4cdc92150ffea11c83e54d8356a88d3e#diff-f62646dcbc99379245eab4c360267b4caad694a07f1d7a7de6ed6153c8999c85R114

File: sonar-go-plugin/src/main/java/org/sonar/go/externalreport/GolangCILintReportSensor.java [hunk 1]
```diff
@@ -63,21 +71,34 @@ public GolangCILintCheckstyleFormatImporter(SensorContext context, String linter
...
+    protected List<Impact> impacts(String severity, String source) {
+      var isSonarCloud = context.runtime().getProduct() == SonarProduct.SONARQUBE && context.runtime().getEdition() == SonarEdition.SONARCLOUD;
+      if (!isSonarCloud) {
+        // SonarQube Cloud does not yet support the `impact` field for external issues
+        if (GOSEC.equals(source)) {
+          return List.of(new Impact(SoftwareQuality.SECURITY, org.sonar.api.issue.impact.Severity.MEDIUM));
+        }
+        return List.of(new Impact(SoftwareQuality.MAINTAINABILITY, org.sonar.api.issue.impact.Severity.MEDIUM));
+      }
+      return List.of();
```
Comment (petertrr) [inline sonar-go-plugin/src/main/java/org/sonar/go/externalreport/GolangCILintReportSensor.java]:
```suggestion
      if (isSonarCloud) {
        // SonarQube Cloud does not yet support the `impact` field for external issues
        return List.of();
      }
      if (GOSEC.equals(source)) {
        return List.of(new Impact(SoftwareQuality.SECURITY, org.sonar.api.issue.impact.Severity.MEDIUM));
      }
      return List.of(new Impact(SoftwareQuality.MAINTAINABILITY, org.sonar.api.issue.impact.Severity.MEDIUM));
```
IMO, early return here would be more redable

**Review by petertrr — APPROVED** [review]

**Comment by mstachniuk** [comment]
The issue requires more complicated solution, the followup epic: https://sonarsource.atlassian.net/browse/SONARGO-76

**Pull request closed without merging**